File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    if(matrix.length <= 0) {
        return false;
    }
    const m = matrix.length;
    const n = matrix[0].length;
    let r = 0, c = n - 1;
    while(r < m && c >=0) {
        let mid = matrix[r][c];
        if(target == mid) {
            return true;
        } else if(target < mid) {
            c--;
        } else {
            r++;
        }
    }
    return false;
};

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if(matrix.length == 0) {
            return false;
        }
        final int m = matrix.length;
        final int n = matrix[0].length;
        int r = 0, c = n-1;
        while(r < m && c >= 0) {
            if(matrix[r][c] == target) {
                return true;
            } else if(target < matrix[r][c]) {
                c--;
            } else {
                r++;
            }
        }
        return false;
    }
}